pkg/skaffold/build/bazel/build.go

```diff
@@ -80,12 +79,11 @@ func (b *Builder) buildTar(ctx context.Context, out io.Writer, workspace string,
 		return "", fmt.Errorf("running command: %w", err)
 	}
 
-	bazelBin, err := bazelBin(ctx, workspace, a)
+	tarPath, err := bazelTarPath(ctx, workspace, a)
 	if err != nil {
-		return "", fmt.Errorf("getting path of bazel-bin: %w", err)
+		return "", fmt.Errorf("getting bazel tar path: %w", err)
 	}
 
-	tarPath := filepath.Join(bazelBin, buildTarPath(a.BuildTarget))
```

Comment:
It looks like this breaks the scenarios where the artifact has a non-default context and the `skaffold` command is executed from within a path different than the repository root.

The `bazel cquery` runs in the directory corresponding to the context and returns the tar path relative to that context but `docker.Push()` still runs in the current working directory and is now supplied with a path relative to the context, not relative to the current working directory.

Take the following example. There is a skaffold file in projects/acme/some-projects:
```yaml
apiVersion: skaffold/v2beta26
kind: Config
build:
  artifacts:
  - image: quay.io/acme/some-project
    context: ../../..
```
If one does:
```
cd projects/acme/some-projects && skaffold build
```
It will no longer work. It will fail with an error like:
```
build [quay.io/acme/some-project] failed: reading image "bazel-out/darwin-fastbuild-ST-4a519fd6d3e4/bin/projects/some-project/docker.tar": open bazel-out/darwin-fastbuild-ST-4a519fd6d3e4/bin/projects/some-project//docker.tar: no such file or directory
```